internal/indexer/indexer.go

```diff
@@ -89,6 +97,13 @@ func (i *Indexer) ProcessTransaction(ctx context.Context, transaction ingest.Led
 			return fmt.Errorf("processing effects state changes: %w", err)
 		}
 		i.Buffer.PushStateChanges(effectsStateChanges)
+
+		// 2.2. Index contract deploy state changes
+		contractDeployStateChanges, err = i.contractDeployProcessor.ProcessOperation(ctx, opParticipants.OpWrapper)
+		if err != nil && !errors.Is(err, processors.ErrInvalidOpType) {
+			return fmt.Errorf("processing contract deploy state changes: %w", err)
+		}
```

Comment:
What if `errors.Is(err, processors.ErrInvalidOpType)` is true?

Reply:
We chose to send ops of different types to multiple processors, and if that processor is not able to process the incoming type (checked in the first lines of its implementation), then the processor returns early with that error.

The alternative would be adding routers or switch/cases to many parts of the code, but that would add unnecessary complexity, given the current code base.

Reply:
Yeah +1 to what Marcelo said. We simply fail the processing and it exits out of the ingestion process.